rotkehlchen/chain/manager.py

```diff
@@ -1569,6 +1574,72 @@ def query_ethereum_balances(
         self.query_ethereum_tokens()
         self._add_protocol_balances()
 
+    @cache_response_timewise()
+    def query_ethereum_lp_balances(
+            self,  # pylint: disable=unused-argument
+            balances: Dict[str, Dict[Asset, Balance]],
+            # Kwargs here is so linters don't complain when the "magic" ignore_cache kwarg is given
+            **kwargs: Any,
+    ) -> Optional['AddressToUniswapV3LPBalances']:
+        """Queries all ethereum liquidity pool balances and adds them to the balances.
+        Returns the Uniswap V3 LP balances as they're needed for the NFTs balances.
+        May raise:
+        - RemoteError if any of the calls fail.
+        """
+        uniswap = self.get_module('uniswap')
+        uniswap_v3_balances = None
+        if uniswap is not None:
+            uniswap_v3_balances = uniswap.get_v3_balances(
+                addresses=self.queried_addresses_for_module('uniswap'),
+            )
+            uniswap_v2_balances = uniswap.get_balances(
+                addresses=self.queried_addresses_for_module('uniswap'),
+            )
+            self.add_ethereum_lp_to_balance(
+                lp_balance=uniswap_v2_balances,
+                balances=balances,
+            )
+
+        sushiswap = self.get_module('sushiswap')
```

Comment:
you still have code repetition here. Come on please don't make me spell it out.

```python
for module_name in ('sushiswap', 'balancer'):
    module = self.get_module(module_name)
    if module is not None:
        module_balances = module.get_balances(
            addresses=self.queried_addresses_for_module(module_name)
        )
        self.add_ethereum_lp_to_balance(
            lp_balance=module_balances,
            balances=balances,
        )
```

Iwould liek to also fit uniswapv2 and v3 in there but ... hm ... adding 2 more ifs in there seems ugly.